core/field_variable.js

```diff
@@ -47,15 +47,24 @@ goog.require('goog.string');
  * @constructor
  */
 Blockly.FieldVariable = function(varname, opt_validator, opt_variableTypes) {
-  Blockly.FieldVariable.superClass_.constructor.call(this,
-      Blockly.FieldVariable.dropdownCreate, opt_validator);
-  this.setValue(varname || '');
+  // The FieldDropdown constructor would call setValue, which might create a
+  // spurious variable.  Just do the relevant parts of the constructor.
+  this.menuGenerator_ = Blockly.FieldVariable.dropdownCreate;
+  this.size_ = new goog.math.Size(0, Blockly.BlockSvg.MIN_BLOCK_Y);
+  this.setValidator(opt_validator);
+  // TODO (#1499): Add opt_default_type to match default value.  If not set, ''.
+  this.defaultVariableName = (varname || '');
+  this.defaultType_ = '';
```

Comment:
The defaultType_ should also take into account what's in the opt_variableTypes list. E.g. if that list has exactly one element, the defaultType_ should be set to that element.

Reply:
Sure, but I'll handle that when I deal with the rest of #1499.